src/lib/homestore_backend/hs_blob_manager.cpp

```diff
@@ -143,11 +148,14 @@ void HSHomeObject::on_blob_put_commit(int64_t lsn, sisl::blob const& header, sis
         RELEASE_ASSERT(iter != _pg_map.end(), "PG not found");
         index_table = static_cast< HS_PG* >(iter->second.get())->index_table_;
         RELEASE_ASSERT(index_table != nullptr, "Index table not intialized");
+        if (iter->second->blob_sequence_num_.load() <= ctx->blob_id_) {
+            iter->second->blob_sequence_num_.store(ctx->blob_id_ + 1);
+        }
     }
 
     BlobInfo blob_info;
     blob_info.shard_id = msg_header->shard_id;
-    blob_info.blob_id = lsn;
+    blob_info.blob_id = ctx->blob_id_;
```

Comment:
ctx can be nullptr when HO restarted and replay the journal log, right?
I think in that case may need to read the blob header from the pba and decode it from the blob header. or put the copy of blob id to the header too.

Reply:
There should be no difference in the commit path when replaying logs versus playing them in real-time. If there is an issue replaying from the log, that issue would also exist on any follower appending/committing in the normal runtime.

Reply:
I am not sure if ctx will be nullptr when follower apply/replay the commited log as we do not have real replication until now. but for HS SoloReplDev,  when it restarts, it will replay all commited log  from the checkpoint LSN in here with ctx is nullptr(https://github.com/eBay/HomeStore/blob/391168223297f672a03fd95e1ddef810535b005a/src/lib/replication/repl_dev/solo_repl_dev.cpp#L117)

Reply:
I am using the attribute `key` to serialize blob id now. This problem should be resolved